### PR TITLE
Rewrite relations query with goqu

### PR DESCRIPTION
**Pull request by SherinV**
Signed-off-by: Sherin Varughese <shvarugh@redhat.com>

For https://github.com/stolostron/backlog/issues/20282
Rewrite relations query with goqu template
~Add relatedKinds filter~

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -7,10 +7,10 @@ package config
```
Comment (jlpadilla) [inline pkg/config/config.go:7]: Thanks for making this consistent with indexer!

File: pkg/resolver/search.go [hunk 2]
```diff
@@ -186,39 +191,50 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
...
+	//The level can be parameterized later, if needed, for applications
+	whereDs = append(whereDs, goqu.C("level").Eq(1)) // Add filter to select only level 1 relationships
 
 	//defining variables
 	items := []map[string]interface{}{}
 	var kindSlice []string
 	var kindList []string
 	var countList []int
 
-	relQuery := strings.TrimSpace(`WITH RECURSIVE
```
Comment (jlpadilla) [inline pkg/resolver/search.go:201]: Should we keep a comment with the original query for documentation?

File: pkg/resolver/search.go [hunk 3]
```diff
@@ -186,39 +191,50 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
...
+					InnerJoin(goqu.T("search_graph").As("sg"),
+						goqu.On(goqu.ExOr{"sg.destid": goqu.I("e.sourceid"), "sg.sourceid": goqu.I("e.destid")})).
+					Select(selectNext...).
+					Where(goqu.Ex{"sg.level": goqu.L("1"),
+						"r.uid": goqu.Op{"neq": goqu.All("{sg.path}")}}))).
+		Select("data", "destid", "destkind").Distinct("destid").
+		Where(whereDs...).ToSQL()
+
+	if err != nil {
+		klog.Error("Error creating relation query", err)
```
Comment (jlpadilla) [inline pkg/resolver/search.go:235]: I wonder if we are able to continue after this error.

File: pkg/resolver/search.go [hunk 4]
```diff
@@ -186,39 +191,50 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
...
+	}
+	relations, relQueryError := s.pool.Query(context.Background(), sql, params...) // how to deal with defaults.
```
Comment (jlpadilla) [inline pkg/resolver/search.go:237]: context.TODO()

File: pkg/resolver/search.go (hunk 2 above)
Comment (SherinV) [reply pkg/resolver/search.go:201]: Done

File: pkg/resolver/search.go (hunk 3 above)
Comment (jlpadilla) [reply pkg/resolver/search.go:255]: Would the client receive an error? GraphQL can send partial results and errors.
Also the error we send to the client shouldn't describe any internals like the SQL query.

**We can move this discussion out of this PR, we should review the error path for the entire project.**

Comment (SherinV) [reply pkg/resolver/search.go:256]: Yes, we will revisit the error path for the entire api along with the context.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**